Activate and inactive chromosome X
=============================================================================

Data generated by `Wang <https://pubmed.ncbi.nlm.nih.gov/29887375/>`_ *et al.* 2019 and featured in `Lappala <https://www.pnas.org/doi/abs/10.1073/pnas.2107092118>`_ *et al.* 2021 were downloaded from the `Gene Expression Omnibus <https://www.ncbi.nlm.nih.gov/geo/>`_ in the ``.summary.txt.gz`` file format. The first wild-type replicate from the record `GSE99991 <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE99991>`_ with accession record `GSM2667262 <https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM2667262>`_ was used in analysis and development of the ``.summary.txt.gz`` to ``.hic`` executable script ( `listed here <https://github.com/4DGB/hic-converter/blob/main/tools/summary.to.chrom.hic.py>`_). Here we store the two haplotype resolved ``.summary.txt.gz`` files for chromosome 13 of the first replicate from Wang *et al.* for the wild-type, cas and mus haplotypes: 

* /summary/GSM2667262_WT1.HiC.rep1.cas.chr13.summary.txt.gz 
* /summary/GSM2667262_WT1.HiC.rep1.mus.chr13.summary.txt.gz

The *cas* and *mus* haplotype names correspond to the activate (Xa) and inactive (Xi) chromosome X states, respectively. The mm9, Mus musculus reference genome was used in analysis.

### Generating a *.hic* file from *.summary.txt.gz* file

Shown below is an example call to the summary to hic command, generating a hic file for the *cas*, activate chromosome X haplotype. 

.. code-block:: console
    ## Change directory to tools
    cd ./hic-converter/tools

    ## Convert a summary.txt.gz file to an .hic file for a single chromosome
    ./summary.to.chrom.hic.py -i ../data/summary/GSM2667262_WT1.HiC.rep1.cas.chr13.summary.txt.gz -g mm9 -c chr13 -O ../data/hic/GSM2667262_WT1.HiC.rep1.cas.chr13.hic


## Hi-C data on mouse embryonic stem cells from Marks *et al.* 

The following code and instructions were modified from a [HicExplorer tutorial]( https://hicexplorer.readthedocs.io/en/latest/content/mES-HiC_analysis.html) on aligning Hi-C data from female mouse, embryonic stem cells. It was adapted to generate a Hi-C contact map for chromosome 13 in *.h5* format. Additionally only data from the first replicate of [Marks *et al.* 2015]( https://genomebiology.biomedcentral.com/articles/10.1186/s13059-015-0698-x) is used in this analysis. See the [main page]( https://github.com/4DGB/hic-converter) of this repository for instruction on setting up a python computing environment and converting the *.h5* file to a *.hic* file.

### Constructing a *.h5* contact map of chromosome 13 from pair-end *.fastq.gz* files

.. code-block:: console
    ## Set the working directory
    cd ./hic-converter/data

    ## Activate the python computing environment.
    conda activate hicexplorerenv

    ## Make directories for analysis.
    mkdir -p genome_mm10 fastq bam bwa

    ## Gather the mm10 reference, use wget and tar to unpack the data.
    wget http://hgdownload-test.cse.ucsc.edu/goldenPath/mm10/bigZips/chromFa.tar.gz -O genome_mm10/chromFa.tar.gz
    tar -xvzf genome_mm10/chromFa.tar.gz

    ## Index chromosome 13 via bwa index.
    bwa index -p bwa/chr13_index genome_mm10/chr13.fa

    ## Use the hicFindRestSite function from HicExplorer to find the SauIII restriction sites (GATC) across the mm10 genome. 
    hicFindRestSite --fasta ./genome_mm10/chr13.fa -o ./genome_mm10/chr13_SauIII_cut_sites.bed --searchPattern GATC

    ## Download data from Marks et al. for alignment.
    wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR195/007/SRR1956527/SRR1956527_1.fastq.gz -O ./fastq/SRR1956527_1.fastq.gz
    wget ftp://ftp.sra.ebi.ac.uk/vol1/fastq/SRR195/007/SRR1956527/SRR1956527_2.fastq.gz -O ./fastq/SRR1956527_2.fastq.gz

    ## Align data seperately for each fastq file to chromosome 13 via bwa mem. Note we do not sort or filter the output bam files.
    bwa mem -A 1 -B 4 -E 50 -L 0 -t 14 bwa/chr13_index ./fastq/SRR1956527_1.fastq.gz | samtools view -Shb - > ./bam/SRR1956527_1_chr13.bam
    bwa mem -A 1 -B 4 -E 50 -L 0 -t 14 bwa/chr13_index ./fastq/SRR1956527_2.fastq.gz | samtools view -Shb - > ./bam/SRR1956527_2_chr13.bam

    ## Construct Hi-C contact matrix in .h5 format via the build matrix function from HicExplorer.
    hicBuildMatrix --samFiles ./bam/SRR1956527_1_chr13.bam ./bam/SRR1956527_2_chr13.bam \
        --restrictionSequence GATC --danglingSequence GATC --restrictionCutFile ./genome_mm10/chr13_SauIII_cut_sites.bed \
        --threads 5 --inputBufferSize 400000 \
        --QCfolder ./SRR1956527_chr13_QC --outFileName ./h5/SRR1956527_chr13.h5